src/plugins/workspace/public/components/workspace_updater/workspace_updater.tsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import { EuiPage, EuiPageBody, EuiPageHeader, EuiPageContent } from '@elastic/eui';
+import { useObservable } from 'react-use';
+import { i18n } from '@osd/i18n';
+import { of } from 'rxjs';
+
+import { WorkspaceAttribute } from 'opensearch-dashboards/public';
+import { useOpenSearchDashboards } from '../../../../../../src/plugins/opensearch_dashboards_react/public';
+
+import { PATHS } from '../../../common/constants';
+import { WorkspaceForm, WorkspaceFormData } from '../workspace_creator/workspace_form';
+import {
+  WORKSPACE_APP_ID,
+  WORKSPACE_ID_IN_SESSION_STORAGE,
+  WORKSPACE_OP_TYPE_UPDATE,
+} from '../../../common/constants';
+import { ApplicationStart } from '../../../../../core/public';
+
+export const WorkspaceUpdater = () => {
+  const {
+    services: { application, workspaces, notifications },
+  } = useOpenSearchDashboards<{ application: ApplicationStart }>();
+
+  const currentWorkspace = useObservable(
+    workspaces ? workspaces.client.currentWorkspace$ : of(null)
+  );
+
+  const excludedAttribute = 'id';
+  const { [excludedAttribute]: removedProperty, ...otherAttributes } =
+    currentWorkspace || ({} as WorkspaceAttribute);
+
+  const [currentWorkspaceFormData, setCurrentWorkspaceFormData] = useState<
```

Comment:
The `useState` will cause extra re-render after `setCurrentWorkspaceFormData` called. How about change to `useRef`?